assignment/final/aLife/code/master.js

```diff
@@ -0,0 +1,417 @@
+            NAME: move.NAME,
+            TYPE: move.TYPE,
+            POWER: move.POWER
+        });
+    }
+    return currentMoves;
+}
+
+function getCurrPkmnStats(pkmn, data) {
+    currentStats = [];
+    statSection.innerHTML =
+        `
+        <div class="col-sm-6">${data.stats[5].stat.name}: ${data.stats[5].base_stat}</div>
+        <div class="col-sm-6">${data.stats[4].stat.name}: ${data.stats[4].base_stat}</div>
+        <div class="col-sm-6">${data.stats[3].stat.name}: ${data.stats[3].base_stat}</div>
+        <div class="col-sm-6">${data.stats[2].stat.name}: ${data.stats[2].base_stat}</div>
+        <div class="col-sm-6">${data.stats[1].stat.name}: ${data.stats[1].base_stat}</div>
+        <div class="col-sm-6">${data.stats[0].stat.name}: ${data.stats[0].base_stat}</div>
+    `
+    currentStats.push({
+            STAT: data.stats[5].stat.name,
+            VALUE: data.stats[5].base_stat
+        }, {
+            STAT: data.stats[4].stat.name,
+            VALUE: data.stats[4].base_stat
+        }, {
+            STAT: data.stats[3].stat.name,
+            VALUE: data.stats[3].base_stat
+        }, {
+            STAT: data.stats[2].stat.name,
+            VALUE: data.stats[2].base_stat
+        }, {
+            STAT: data.stats[1].stat.name,
+            VALUE: data.stats[1].base_stat
+        }, {
+            STAT: data.stats[0].stat.name,
+            VALUE: data.stats[0].base_stat
+        },
+
+    );
+    return currentStats;
+}
+
+function getPkmn(x) {
+    //console.log("running in getPkmn(new pkmn)");
+    let pkmn = pkmnList.find(pkmn => pkmn.NAME === x);
+
+    fetch(baseUrl + 'pokemon/' + pkmn.ID + '/')
+        .then(res => res.json())
+        .then(data => {
+            //console.log(data);
+            pkmn.TYPE1 = data.types[0].type.name;
+            pkmn.TYPE2 = data.types.length > 1 ? data.types[1].type.name : "";
+            pkmn.SPRITES = data.sprites;
+            type1.innerHTML = pkmn.TYPE1;
+            type2.innerHTML = pkmn.TYPE2;
+            idBox.innerHTML = pkmn.ID;
+            terminalScreen.innerHTML =
+                `
+                <img src="${data.sprites.front_default}">
+                `
+            pkmn.STATS = getCurrPkmnStats(pkmn, data);
+            pkmn.MOVES = getCurrPkmnMoves(pkmn, data);
+        });
+    currentPkmn = pkmn;
+    return pkmn
+}
+
+function selectNewPkmn(x) {
+    const regex = new RegExp(".*" + pkmnName.value + ".*");
+    let count = pkmnList.filter(pkmn => pkmn.NAME.match(regex)).length;
+    //console.log(pkmnListForTypeFilter);
+    let count2 = pkmnList.length;
+    if(pkmnListForTypeFilter.length == 0){
+        for (let i = 0; i < count; i++) {
+            document.getElementById(pkmnList.filter(pkmn => pkmn.NAME.match(regex))[i].NAME).classList.contains("pkmn") ?
+            document.getElementById(pkmnList.filter(pkmn => pkmn.NAME.match(regex))[i].NAME).classList.remove("pkmn") : "";
+        }
+    }
+    else{
+        try{
```

Comment:
is this try in case `pkmnListForTypeFilter` is not defined...or does not have a length?

Reply:
I “think” it’s because some pokemon.pokemon.name don’t exist in the national Pokédex so this is a way around it.